app/modules/code_provider/github/github_router.py

```
@@ -12,24 +12,57 @@
 
 @router.get("/github/user-repos")
 async def get_user_repos(
-    user=Depends(AuthService.check_auth), db: Session = Depends(get_db)
+    search: str = Query(None, description="Search query to filter repositories"),
+    user=Depends(AuthService.check_auth), 
```

Ruff (0.15.0)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)